NewForce-Cohort-5/tabloidmvc-the-spaghetti-squad
# Hb-tkt15-delete

# Description

An admin user should be able to see and click on the Tag Management link in the menu to get the list of tags. 
The user will see a list of tags that are in alphabetical order.  
There are create, edit, and delete options.  These create and edit options will not work at this time. 
However the delete option will give the user an option to delete or reject confirmation.
If the user chooses delete they will be sent back to the tag list and the tag they deleted is removed. 
If the user chooses to reject the delete confirmation by clicking return to list. The user will be returned to the list and the tag will not be deleted. 


I added a GetTagById and DeleteTag Method to the tag repositories and created a view for the delete method in view. 

There are no permissions for this ticket but will be included in future tickets.  Assume all users are admins. 


Related Ticket(s):
[https://trello.com/c/aOFZ3YR5]


## Type of change



# Testing Instructions


1. After verifying that you have our repo cloned down into a local repository, `git fetch -a ` in the main branch to retrieve the named **hb-tkt15Delete** that I have pushed up
2. In Terminal `git checkout hb-tkt15Delete` to switch to my branch and in the terminal type `Start TabloidMVC.sln` to open in Visual Studios 2022. 
3. If you have not seeded your database with this data, seed your data  from SQLQuery link : [(https://github.com/NewForce-Cohort-5/tabloidmvc-the-spaghetti-squad/tree/main/SQL ](https://github.com/NewForce-Cohort-5/tabloidmvc-the-spaghetti-squad/tree/main/SQL )
4. You will need to create the table first and then add the seed data. Choose files in the following order:
         -[01_TabloidMVC_Create_DB.sql](https://github.com/NewForce-Cohort-5/tabloidmvc-the-spaghetti-squad/blob/main/SQL/01_TabloidMVC_Create_DB.sql)
          - [02_TabloidMVC_Seed_Data.sql](https://github.com/NewForce-Cohort-5/tabloidmvc-the-spaghetti-squad/blob/main/SQL/02_TabloidMVC_Seed_Data.sql)
7. Now let's add some test seed data for tags by going to the TabloidMVC database and clicking new query and inserting the following code then hit green play button, this add give you two tags to delete:  
```
 Insert into Tag 
(Name)
 VALUES ('Delete Test 1');

  Insert into Tag 
(Name)
 VALUES ('Delete Test 2');

select * from Tag;
```

9. Now that you have your seed data, we can test.
10. Click the solid green play button and you will arrive at the home page. You will need to login as [admin@example.com](mailto:admin@example.com).
11. Next go to Tag Management in the menu. This is where you can see the list of tags in alphabetical order and the delete option.  
12. Please test at least 2 times by choosing the tags we added to delete and it will return you back to the list showing the tag selected was removed. 
13. Also reject the delete to be sure it goes back to list and the tag has not remove tag. 


 

# Checklist:
